crates/sui-core/src/authority.rs

```diff
@@ -498,6 +504,12 @@ impl AuthorityMetrics {
                 registry,
             )
             .unwrap(),
+            execution_queueing_delay_ms: register_int_gauge_with_registry!(
+                "execution_queueing_delay_ms",
+                "Average queueing delay between a transaction is ready for execution until it starts executing.",
+                registry
+            )
+            .unwrap(),
```

Comment:
should this be a histogram?

Reply:
Good question. I wasn't paying attention to other metrics, and only wanted this metrics to match the LatencyObserver value for debugging purpose. But looks like all the latencies are histogram, so I switched to Histogram. But I still do the metrics update inside sampling so that the value can still match what LatencyObserver sees. Don't have strong opinion if we need to move the metric updates outside of the sampling.

Reply:
I don't think we necessarily need to move it out of the sampling code

Reply:
Current metric updates are still inside sampling.